### PR TITLE
Ban feature

### PR DESCRIPTION
**Pull request by greatman**
Added ban system using the spout.yml file.

### CODE REVIEW

File: src/main/java/org/spout/server/SpoutServer.java [hunk 1]
```diff
@@ -787,17 +788,66 @@ public World getDefaultWorld() {
...
-		// TODO Auto-generated method stub
-
+	public void ban(Player player) {
+		
+		bannedPlayers.add(player.getName());
+		Configuration config;
+		
+		try
+		{
+			config = getConfiguration(true);
```
Comment (Afforess) [inline src/main/java/org/spout/server/SpoutServer.java]: Is there a reason you need to FORCE a configuration update with that true parameter?

File: src/main/java/org/spout/server/SpoutServer.java [hunk 2]
```diff
@@ -787,17 +788,66 @@ public World getDefaultWorld() {
...
+			
+			if (bannedList == null) {
+				bannedList = new ArrayList<String>();
+			}
+			
+			if (!bannedList.contains(player.getName())) {
+				bannedList.add(player.getName());
+			}
+				
+			config.setProperty("banlist", bannedList);
```
Comment (Afforess) [inline src/main/java/org/spout/server/SpoutServer.java]: You don't need to set the property unless it was null.

File: src/main/java/org/spout/server/SpoutServer.java [hunk 3]
```diff
@@ -787,17 +788,66 @@ public World getDefaultWorld() {
...
+			config.save();
```
Comment (Afforess) [inline src/main/java/org/spout/server/SpoutServer.java]: Is there a reason we need to save this right away? Wouldn't saving at shutdown be acceptable? File IO is terribly slow.

File: src/main/java/org/spout/server/SpoutServer.java [hunk 4]
```diff
@@ -787,17 +788,66 @@ public World getDefaultWorld() {
...
+	public void unban(Player player) {
 		// TODO Auto-generated method stub
-
+		String playerName = player.getName();
+		bannedPlayers.remove(playerName);
+		Configuration config;
+		
+		try
+		{
+			config = getConfiguration(true);
```
Comment (Afforess) [inline src/main/java/org/spout/server/SpoutServer.java]: Same comments as above.

File: src/main/java/org/spout/server/SpoutServer.java (hunk 2 above)
Comment (greatman) [inline src/main/java/org/spout/server/SpoutServer.java]: So basicly just add into the bannedList variable and the server takes care of the rest?

Comment (Afforess) [inline src/main/java/org/spout/server/SpoutServer.java]: Right. The configuration reference will be up to date still. You don't need to recheck the file every time.

**Comment by Salynne** [comment]
Talked about this a bit on IRC and came to some decisions.

1) I don't agree that it should wait til shutdown to save. Afforess may come back and override me on that so don't worry about that right now. The reasoning is that you do not save very often and you do not want to crash and lose bans. Banned people coming back is typically bad.

2) I would like to see IP bans still being possible alongside player bans, this prevents stuff like some crappy DDOS-Style stuff I have been hit with before, among other things.

3) The ban/whitelist/iplist should not be in the main spout.yml. They should be in their own individual yaml or txt file.

I at least want 3 to exist to pull, if you can do 2 for me that'd be great, and we'll have to wait to see on 1 because I don't want to reverse his decision and have you change it and then have him reverse it again. THat'd be silly :P

**Comment by Afforess** [comment]
> 1) I don't agree that it should wait til shutdown to save. Afforess may come back and override me on that so don't worry about that right now. The reasoning is that you do not save very often and you do not want to crash and lose bans. Banned people coming back is typically bad.

Yeah I wanted 300 second saving intervals.

**Comment by greatman** [comment]
I can work out 2 and 3. Ill let you guys work on 1 because I know I will
have a bad implementation lol.

**Comment by Salynne** [comment]
Closing these, open a new one if you've updated them.

**Pull request closed without merging**